src/map.rs

```diff
@@ -951,40 +951,41 @@ where
         let c = new_capacity as isize; // TODO
 
         loop {
-            let sc = self.size_ctl.load(Ordering::SeqCst);
+            let mut sc = self.size_ctl.load(Ordering::SeqCst);
             if sc < 0 {
                 break;
             }
 
-            let table = self.table.load(Ordering::SeqCst, &guard);
+            let mut table = self.table.load(Ordering::SeqCst, &guard);
```

Comment:
I don't think this has to be mutable? You never use the value later, so it doesn't need to be re-assigned.

Reply:
Right, I did it this way for the moment, since that is what the java version did, but you're totally correct in that we can just bind to a new variable. I'll change this in a bit.

Reply:
Ah, but keep in mind that in the Java version, `table` refers to the equivalent of `self.table`, it is _not_ a local variable!

Reply:
You're right, I still get confused with the automatic `this.` sometimes, haven't done too much java.
This shouldn't be a problem, since our `table` is the java implementations `tab` which is a local and `new_table` is `nt` which is also a local.

What is your opinion on variable names btw.? I would like to give the variables more expressive names, but would also be okay with staying as close to the java version as possible...

Reply:
Hehe, yeah, it can definitely get confusing. You're right that the code is still correct, but when writing unsafe code like this, I like to not mutate pointers or keep them around any more than I have to. In this case, there is no reason for the (local) variable `table` to be `mut`, and therefore it shouldn't be. It's just to prevent ourselves from doing something stupid down the line :sweat_smile: 

As for variable names, I'm totally fine with _either_ using the exact same variable names as the Java code, or coming up for _good_ descriptive names. The only thing I wouldn't like is giving a variable a different but still relatively obscure/generic name :)